mne/utils.py

```diff
@@ -790,6 +790,137 @@ def dec(*args, **kwargs):
     return dec
 
 
+def copy_doc(source):
+    """Decorator to copy the docstring from another function.
+
+    This is useful when inheriting from a class and overloading a method. This
+    decorator can be used to copy the docstring of the original method:
+
+        >>> class A:
+        ...     def m1():
+        ...         "Docstring for m1"
+        ...         pass
+        >>> class B (A):
+        ...     @copy_doc(A.m1)
+        ...     def m1():
+        ...         pass
+        >>> print(B.m1.__doc__)
+        Docstring for m1
+
+    Parameters
+    ----------
+    source : function
+        Function to copy the docstring from
+
+    Returns
+    -------
+    wrapper : function
+        The decorated function
+    """
+    def wrapper(func):
+        func.__doc__ = source.__doc__
+        return func
+    return wrapper
+
+
+def copy_function_doc_to_method_doc(source):
+    """Use the docstring from a function as docstring for a method.
+
+    This decorator is useful when implementing a method that just calls a
+    function. This pattern is prevalent in for example the plotting functions
+    of MNE. The first parameter specified in the docstring of the function is
+    removed in the docstring for the method (it is assumed to be set to
+    `self`):
+
+        >>> def plot_function(object, a, b):
+        ...     '''Docstring for plotting function.
+        ...     Parameters
+        ...     ----------
+        ...     object : instance of object
+        ...         The object to plot
+        ...     a : int
+        ...         Some parameter
+        ...     b : int
+        ...         Some parameter'''
+        ...     pass
+        ...
+        >>> class A:
+        ...     @copy_function_doc_to_method_doc(plot_function)
+        ...     def plot(self, a, b):
+        ...         plot_function(self, a, b)
+        >>> print(A.plot.__doc__)
+        Docstring for plotting function.
+            Parameters
+            ----------
+            a : int
+                Some parameter
+            b : int
+                Some parameter
+
+    Parameters
+    ----------
+    source : function
+        Function to copy the docstring from
+
+    Returns
+    -------
+    wrapper : function
+        The decorated method
+    """
+    def wrapper(func):
+        doc = source.__doc__.split('\n')
+
+        # Find parameter block
+        for line, text in enumerate(doc[:-2]):
+            if (text.strip() == 'Parameters' and
+                    doc[line + 1].strip() == '----------'):
```

Comment:
what if there is a typo here or the number of hyphens is somehow less? This will pass silently and you will incorrectly assume that there is no parameter block